### PR TITLE
Update 1.10 enterprise upgrade docs.

### PR DESCRIPTION
**Pull request by mkeeler**
I think I have some overall structure in place but I found out that our [general upgrade guide](https://www.consul.io/docs/upgrading/instructions/general-process) is kind of lacking. It goes over how to do an in place upgrade of OSS but doesn't cover autopilot assisted upgrades, client upgrades or snapshot agent upgrades.

Additionally, my plan was to have version specific upgrade paths in the guide. However I realized that clusters could have mixed versions and then specifying that would get way too complicated vs just saying that at each step all servers, or clients or snapshot agents needed to be upgraded to the latest in either the 1.8.x or 1.9.x series.

There are a bunch of vague steps in the 1.10 upgrade guide that I want to fill in with more info but was hoping for others opinions on how to proceed. Originally I was going to link out to the general upgrade process but that just covers upgrading the servers using an in-place upgrade strategy. This didn't seem great as I expect many enterprise users are going to use automated upgrades. So basically what would others like to see at these locations. It feels like we should have other documentation around the various scenarios but in the absence of that I am not sure what would be best in these docs.

### CODE REVIEW

**Comment by mkeeler** [comment]
The current 1.10 upgrade page instructions are live here: https://consul-7a2vaizou-hashicorp.vercel.app/docs/upgrading/instructions/upgrade-to-1-10-x

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+# Upgrading to 1.10.0
+
+<EnterpriseAlert />
+
+## Introduction
+
+This guide explains how to best upgrade a single Consul Enterprise datacenter to v1.10.0
+from a version of Consul that is forwards compatible with v1.10. If you are on a major 
+version of Consul prior to 1.8, please follow the link for the version you are on from 
+[here](/docs/upgrading/instructions). If you are already on major versions 1.8 or 1.9
```
Comment (banks) [inline website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:18]: I wonder if this is clear enough that they should follow that other guide _and then come back_ and complete this one? Perhaps something like:

```suggestion
version of Consul prior to 1.8, you will need to complete and [upgrade to 1.8.13](/docs/upgrading/instructions) or higher before continuing with this guide. If you are already on major versions 1.8 or 1.9
```

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+multiple Consul datacenters then they are upgraded in the normal order with each
+upgrade taking into account the instructions below.
+
+For the open source version of Consul please follow the 
+[General Upgrade Process](/docs/upgrading/instructions/general-process).
+
+~> You can only upgrade to Consul Enterprise 1.10 from a version of Consul Enterprise 
+1.8 >= 1.8.13 or 1.9 >= 1.9.7. Other versions of Consul Enterprise are not forwards
+compatible with v1.10 and will cause issues during the upgrade that could result in
+agents failing to start.
```
Comment (banks) [inline website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:29]:
```suggestion
compatible with v1.10 and will cause issues during the upgrade that could result in
client and snapshot agents failing to start due to [changes in the way we manage licenses](/??).
```
Not sure offhand where that link should go but maybe the FAQ on learn?

Comment (banks) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:29]: Also it would be worth adding something like this to the callout box:

> This guide will walk through upgrading first to a compatible patch release and then completing the 1.10 upgrade from there.

Without that, I even was confused that right after saying "you need 1.8.13" the next paragraph below says " you need >= 1.8.0" before I realised why the discrepancy.

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+## Requirements
+
+- All Consul servers, clients and snapshot agents should be on a version of Consul >= 1.8.0 and < 1.10.0.
+
+## Assumptions
+
+This guides makes the following assumptions:
+
+- You are familiar with the [General Upgrade Process](/docs/upgrading/instructions/general-process).
+- You have the ability to run Consul CLI commands.
```
Comment (banks) [inline website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:40]: Do we need to state that they also have an operator token?

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+This guides makes the following assumptions:
+
+- You are familiar with the [General Upgrade Process](/docs/upgrading/instructions/general-process).
+- You have the ability to run Consul CLI commands. 
+
+## Considerations
+
+The main breaking changes in Consul Enterprise 1.10 that needs special handling during upgrading are
+the licensing changes outlined on the [Specific Version Details](/docs/upgrading/upgrade-specific#consul-1-10-0)
+page. There are notes on that paged about other changes that might cause issues during an upgrade as well.
```
Comment (banks) [inline website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:46]:
```suggestion
page. There are notes on that page about other changes that might cause issues during an upgrade as well.
```

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+You can find more granular details in the full [changelog](https://github.com/hashicorp/consul/blob/release/release/1.10.0/CHANGELOG.md#1100-june-22-2021).
+Looking through these changes prior to upgrading is highly recommended.
```
Comment (banks) [inline website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:48]: Worth linking the FAQ too?

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+Error querying agent: Unexpected response code: 403 (Permission denied)
+```
+
+```text
+        acl = enabled
+
+```
+
+<Tabs>
+<Tab heading="ACLs Enabled">
```
Comment (banks) [inline website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:79]: I like the tabs approach which limits the apparent complexity. Perhaps we should make double sure this is easy to follow with a sentence above like:

> Select the correct tab below to see the steps based on whether you have ACLs enabled of disabled in your environment.

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+
+**3.** Upgrade all server agents to the latest 1.8.x or 1.9.x release.
```
Comment (banks) [inline website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:81]:
```suggestion
**3.** Upgrade all server agents to the latest 1.8.x or 1.9.x release. As with standard upgrades you should follow our [General Update Process](/docs/upgrading/instructions/general-process)
``` 

maybe the same below just to be totally clear?

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+
+**7.** Upgrade all server agents to v1.10.0
+
+**8.** Upgrade all client agents to v1.10.0
+
+**9.** Upgrade all snapshot agents to v1.10.0
+
+</Tab>
+<Tab heading="ACLs Disabled">
+
```
Comment (banks) [inline website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:102]: Is it worth an introductory sentence here pointing out that we don't recommend running in production with ACLs disabled and that an alternative path is to work on enabling ACLs prior to upgrading?

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+**3.** Upgrade all server agents to the latest 1.8.x or 1.9.x release.
+
+**4.** Pre-configure the client agents with a license.
+
+This can either be set with the `license_path` configuration item, the 
+`CONSUL_LICENSE_PATH` environment variable or the `CONSUL_LICENSE` 
+environment variable. See the [licensing documentation](/docs/enterprise#licensing) 
+for more information about how to configure the license.
```
Comment (banks) [inline website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:110]: Should we note that if they are adding this to the agent config file, they need to be careful to complete the upgrade immediately otherwise client agents will fail to parse the config on restarts if they aren't upgraded yet.

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+
+**5.** Upgrade all client agents to the latest 1.8.x or 1.9.x release.
```
Comment (banks) [inline website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:112]: feels worth calling out the normal processes for checking breaking changes and taking snapshots etc must be followed here?

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+
+**6.** Pre-configure the client agents with a license.
```
Comment (banks) [inline website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:114]: This is the same as step #4 ? Should it be here or there?

Comment (banks) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:114]: Or was this meant to read "snapshot agents"?

File: website/content/docs/upgrading/upgrade-specific.mdx [hunk 12]
```diff
@@ -35,18 +35,7 @@ agents require that either the [`start_join`](/docs/agent/opts#start_join) or
 agents. If those stipulations are not met, attempting to start the client or snapshot agent will
 result in it immediately shutting down.
```
Comment (banks) [inline website/content/docs/upgrading/upgrade-specific.mdx:36]: Is it worth including a link to our FAQ page somewhere in here?

**Review by banks — COMMENTED** [review]
Looks super close actually, these are not major changes but I think a few might be important in keeping things as clear as possible!

File: website/content/docs/upgrading/instructions/index.mdx [hunk 13]
```diff
@@ -29,6 +29,8 @@ To get instructions for your upgrade, please choose the release series you are _
 - [1.5.x](/docs/upgrading/instructions/upgrade-to-1-6-x)
 - [1.6.x](/docs/upgrading/instructions/upgrade-to-1-8-x)
 - [1.7.x](/docs/upgrading/instructions/upgrade-to-1-8-x)
+- [1.8.x](/docs/upgrading/instructsion/upgrade-to-1-10-x)
```
Comment (preetapan) [inline website/content/docs/upgrading/instructions/index.mdx:32]: typo, should be `instructions`

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx [hunk 14]
```diff
@@ -0,0 +1,140 @@
...
+compatible with v1.10 and will cause issues during the upgrade that could result in
+agents failing to start.
+
+## Requirements
+
+- All Consul servers, clients and snapshot agents should be on a version of Consul >= 1.8.0 and < 1.10.0.
+
+## Assumptions
+
+This guides makes the following assumptions:
```
Comment (preetapan) [inline website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:37]: I would label this section header `#Prerequisites` and then list the prerequisites below.

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx [hunk 15]
```diff
@@ -0,0 +1,140 @@
...
+```
+
+<Tabs>
+<Tab heading="ACLs Enabled">
+
+**3.** Upgrade all server agents to the latest 1.8.x or 1.9.x release.
+
+**4.** Upgrade all client agents to the latest 1.8.x or 1.9.x release.
+
+**5.** Upgrade all snapshot agents to the latest 1.8.x or 1.9.x release.
```
Comment (preetapan) [inline website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:85]: maybe link to https://www.consul.io/commands/snapshot/agent for the "Snapshot agents" phrase?

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx [hunk 16]
```diff
@@ -0,0 +1,140 @@
...
+## Procedures
+
+**1.** Retrieve the datacenter's current license by executing the following CLI command:
+
+```shell
+consul license get -signed
+```
+
+This will be useful to have on hand when preparing the license for the 1.10 upgrades later
+on in the process.
```
Comment (preetapan) [inline website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:59]: Does this save the license in a file named LICENSE? Would be good to explicitly state that since the rest of this guide is very specific and operators can follow the instructions to the T.

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx (hunk 8 above)
Comment (preetapan) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:102]: :+1 ,  I think its worth stating that and linking to our ACL guidel

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx (hunk 3 above)
Comment (mkeeler) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:39]: Maybe a conditional requirement if ACLs are in use?

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx [hunk 17]
```diff
@@ -0,0 +1,139 @@
...
+
+**9.** Upgrade all server agents to v1.10.0
+
+**10.** Upgrade all client agents to v1.10.0
+
+**11.** Upgrade all snapshot agents to v1.10.0
+</Tab>
+</Tabs>
+
+## Post-Upgrade Configuration Changes
```
Comment (preetapan) [inline website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:137]: Let's remove this section, its confusing to add it and then say nothing further is required.

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx (hunk 14 above)
Comment (mkeeler) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:36]: That would make more sense to me. However this format seemed standardized across all the upgrade instructions linked on this page https://www.consul.io/docs/upgrading/instructions.

Do you still think its worthwhile deviating in just this doc? Maybe its a change we want to make across all of them?

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx (hunk 11 above)
Comment (mkeeler) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:113]: It was meant to read snapshot agents.

File: website/content/docs/upgrading/upgrade-specific.mdx (hunk 12 above)
Comment (mkeeler) [reply website/content/docs/upgrading/upgrade-specific.mdx:36]: Yep

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx (hunk 6 above)
Comment (thisisnotashwin) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:78]: Does it make sense to have a separate tab for Kubernetes? The upgrade path for Kubernetes will read slightly differently and I can either add it here as a suggestion or branch off of this PR branch and document it there.

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx (hunk 17 above)
Comment (mkeeler) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:137]: It seemed part of the "template" for these pages. I agree it does not seem useful but this page about upgrading to 1.8.x has the exact same bit about nothing extra being required: https://www.consul.io/docs/upgrading/instructions/upgrade-to-1-8-x

**Comment by preetapan** [comment]
Added some comments. To your question around : 
>There are a bunch of vague steps in the 1.10 upgrade guide that I want to fill in with more info but was hoping for others opinions on how to proceed. Originally I was going to link out to the general upgrade process but that just covers upgrading the servers using an in-place upgrade strategy. This didn't seem great as I expect many enterprise users are going to use automated upgrades. So basically what would others like to see at these locations. It feels like we should have other documentation around the various scenarios but in the absence of that I am not sure what would be best in these docs.

IMO, we have existing docs around in place upgrades vs autopilot upgrades already. So I'd add a generic sentence before the detailed steps in the preceding paragraph, something like "These instructions require licensing related configuration changes needed to upgrade to 1.10, and also require intermediate upgrade versions to ensure forward compatibility. We recommend using your existing upgrade strategies such as <link to autopilot upgrade guide> during this multi stage process"

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx (hunk 16 above)
Comment (mkeeler) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx]: I changed the command to save a `consul.hclic` file.

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx (hunk 14 above)
Comment (preetapan) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:36]: ah I didn't realize this was some shared format, ok to leave this as is for now, it was a small nit.

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx (hunk 17 above)
Comment (preetapan) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:144]: same as above. fine by me to leave it as is if its a shared format.

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx (hunk 6 above)
Comment (thisisnotashwin) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:89]: We wouldnt need a separate tab for ACLs enabled vs disabled for Kubernetes as the helm chart manages that complexity under the hood for the user. They only need to provide us with the license file and we ensure we mount it to the right components depending on their existing configuration.

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx (hunk 10 above)
Comment (mkeeler) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:126]: Specifically here? We call out checking for breaking changes in the Considerations section already.

I can add the bit about snapshotting but where should it be. I could see us advocating for taking a snapshot at each point where versions are stable (pre-upgrade and 1.8.13/1.9.7 upgrade is complete across servers, clients and snapshot agents)

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx (hunk 9 above)
Comment (mkeeler) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx]: I added something noting there intended use, that we don't support online reloading and that it is important to promptly proceed with the upgrade and not leave agents in this state forever.

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx [hunk 18]
```diff
@@ -0,0 +1,151 @@
...
+**8.** Pre-configure the server agents with a license. This is the same process that
+was executed for the servers in step 4.
+
+**9.** Upgrade all server agents to v1.10.0.
+
+**10.** Upgrade all client agents to v1.10.0.
+
+**11.** Upgrade all [snapshot agents](/docs/commands/snapshot/agent) to v1.10.0.
+
+</Tab>
```
Comment (thisisnotashwin) [inline website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:146]: Tentative Kubernetes Docs
```suggestion
</Tab>
<Tab heading="Kubernetes">
  
**3.** Upgrade to the latest Consul helm chart v0.32.0+.

**4.** Update the value of `global.image` in the values file to the latest 1.8.x or 1.9.x image.
Additionally, ensure that the Kuberenetes secret with the license is correctly specified in the
values `server.enterpriseLicense.secretName` and `server.enterpriseLicense.secretKey`.

**5.** Upgrade the cluster.
  
Follow the steps in the [Kubernetes Upgrade Guide](docs/k8s/upgrade#upgrade-consul-on-kubernetes)
to then upgrade the cluster. Ensure you check the steps to upgrade the [servers](docs/k8s/upgrade#upgrading-consul-servers)
and the [clients](https://www.consul.io/docs/k8s/upgrade#upgrading-consul-clients).

**6.** Update the value of `global.image` in the values file to the 1.10.0 image.

**7.** Upgrade the cluster.

Follow the steps in the [Kubernetes Upgrade Guide](docs/k8s/upgrade#upgrade-consul-on-kubernetes)
to then upgrade the cluster. Ensure you check the steps to upgrade the [servers](docs/k8s/upgrade#upgrading-consul-servers)
and the [clients](https://www.consul.io/docs/k8s/upgrade#upgrading-consul-clients).
  
</Tab>
```

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx (hunk 2 above)
Comment (mkeeler) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx]: I think I addressed both bits of this. I tried to make it clearer that this guide can really only be used if you are using versions 1.8+ and if you are not you should follow the normal upgrade procedures to get there first.

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx (hunk 7 above)
Comment (mkeeler) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:94]: Do you think we still need this now that I added a callout at the very beginning of the Procedures section?

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx (hunk 18 above)
Comment (thisisnotashwin) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:176]: https://github.com/hashicorp/consul/pull/10449

Comment (mkeeler) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:176]: What about if only clients are in k8s?

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx [hunk 19]
```diff
@@ -38,25 +38,33 @@ This guides makes the following assumptions:
...
-You can find more granular details in the full [changelog](https://github.com/hashicorp/consul/blob/release/release/1.10.0/CHANGELOG.md#1100-june-22-2021).
+You can find more granular details in [FAQ](/docs/enterprise/license/faq) as well as the full 
+[changelog](https://github.com/hashicorp/consul/blob/release/release/1.10.0/CHANGELOG.md#1100-june-22-2021).
 Looking through these changes prior to upgrading is highly recommended.
 
 ## Procedures
 
+~> When any of the steps below call for upgrading components, we assume you follow
+the normal safe upgrade procedure you normally use for upgrades. The basic server
+upgrade procedure is documented [here](/docs/upgrading/instructions/general-process).
```
Comment (preetapan) [inline website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx]: Nit, instead of linking to the word "here" suggest rewording the sentence to "Refer to our documentation for [basic server upgrade process](), and for [autopilot assisted upgrades]()"

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx (hunk 10 above)
Comment (mkeeler) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx]: Given our conversation I think we decided it is okay as is after my modifications to add in information about when to take snapshots.

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx (hunk 19 above)
Comment (mkeeler) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx]: Done.

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx [hunk 20]
```diff
@@ -0,0 +1,222 @@
...
+- You have the ability to run Consul CLI commands. 
+- If ACLs are in use then you possess a token with at least `operator:read` permissions.
+
+## Considerations
+
+The main breaking changes in Consul Enterprise 1.10 that need special handling during upgrading are
+the licensing changes outlined on the [Specific Version Details](/docs/upgrading/upgrade-specific#consul-1-10-0)
+page. There are notes on that page about other changes that might cause issues during an upgrade as well.
+You can find more granular details in the [licensing FAQ](/docs/enterprise/license/faq) as well as the full 
+[changelog](https://github.com/hashicorp/consul/blob/master/CHANGELOG.md#1100-june-22-2021).
```
Comment (mkeeler) [inline website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:51]: Note that this link will be broken until we release 1.10

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx [hunk 21]
```diff
@@ -0,0 +1,222 @@
...
+Looking through these changes prior to upgrading is highly recommended.
+
+## Procedures
+
+~> When any of the steps below call for upgrading components, we assume you follow the normal safe upgrade
```
Comment (preetapan) [inline website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:56]: nit, suggested re-wording 
Any steps in the following section that mention upgrading servers/clients assume that normal safe upgrade procedures are followed. ..

**Review by preetapan — APPROVED** [review]

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx (hunk 21 above)
Comment (mkeeler) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx]: 👍

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx [hunk 22]
```diff
@@ -0,0 +1,222 @@
...
+<Tabs>
+<Tab heading="ACLs Enabled">
+
+**4.** Upgrade all server agents to the latest 1.8.x or 1.9.x release.
+
+**5.** Upgrade all client agents to the latest 1.8.x or 1.9.x release.
+
+**6.** Upgrade all [snapshot agents](/docs/commands/snapshot/agent) to the latest 1.8.x or 1.9.x release.
+
+**7.** Take a snapshot of the upgraded cluster by running:
```
Comment (banks) [inline website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:110]: nit: where should they run that, on a server? Should we mention checking disk space before running it? That gets technical but maybe at least a nod towards the fact this could write a large file to disk might be useful?

**Review by thisisnotashwin — APPROVED** [review]
thanks Matt!!

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx [hunk 23]
```diff
@@ -0,0 +1,222 @@
...
+  
+  This will require pre-configuring the servers with a license
+  before running 1.10. This can either be set with the `license_path` configuration item, the 
+  `CONSUL_LICENSE_PATH` environment variable or the `CONSUL_LICENSE` 
+  environment variable. See the [licensing documentation](/docs/enterprise/license/overview) 
+  for more information about how to configure the license. Note that while
+  Consul Enterprise 1.8.13 and 1.9.7 have the ability to load the license
+  from these configurations, it is intended to only be used during upgrades.
+  Licenses loaded this way are not online reloadable. It is therefore important
+  to promptly proceed with the upgrade and not leave agents in the intermediate state.
```
Comment (banks) [inline website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:207]: Should this block be here in the Kube guide @thisisnotashwin? It seems odd talking about ENV vars when the use only needs to set things in helm values?

Comment (banks) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:207]: never mind this is for external servers case

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx (hunk 22 above)
Comment (mkeeler) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx]: I added information that said they could be run from pretty much anywhere but if you don't execute from a server/client then you will need to pass other options to the save command (and linked out to it). Also added a second note about disk requirements.

File: website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx (hunk 6 above)
Comment (lkysow) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:115]: > They only need to provide us with the license file and we ensure we mount it to the right components depending on their existing configuration.

In fact users will already have a license file in a secret so they only need to bump the Consul image and then run `helm upgrade`. Basically we should copy this text: https://www.consul.io/docs/enterprise/license/faq#kubernetes somewhere in here.

Comment (thisisnotashwin) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:115]: @lkysow We created a tab for Kubernetes and have described just that. We could add the faq steps there as well. The only additional thing we added were steps to ensure people who had consul servers external to kubernetes had steps that they could follow as well. https://consul-m514xwknh-hashicorp.vercel.app/docs/upgrading/instructions/upgrade-to-1-10-x

Comment (lkysow) [reply website/content/docs/upgrading/instructions/upgrade-to-1-10-x.mdx:115]: perfect

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/393397.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 76857dd682ccb77dd30489637a2ecde9193ff14d onto `release/1.10.x` succeeded!
